examples/preprocessing/plot_muscle_detection.py

```diff
@@ -0,0 +1,80 @@
+"""
+===========================
+Annotate muscle artifacts
+===========================
+
+Muscle contractions produce high frequency activity that can mask brain signal
+of interest. Muscle artifacts can be produced when clenching the jaw,
+swallowing, or twitching a cranial muscle. Muscle artifacts are most
+noticeable in the range of 110-140 Hz.
+
+This example uses :func:`~mne.preprocessing.annotate_muscle_zscore` to annotate
+segments where muscle activity is likely present. This is done by band-pass
+filtering the data in the 110-140 Hz range. Then, the envelope is taken using
+the hilbert analytical signal to only consider the absolute amplitude and not
+the phase of the high frequency signal. The envelope is z-scored and summed
+across channels and divided by the square root of the number of channels.
+Because muscle artifacts last several hundred milliseconds, a low-pass filter
+is applied on the averaged z-scores at 4 Hz, to remove transient peaks.
+Segments above a set threshold are annotated as ``BAD_muscle``. In addition,
+the ``min_length_good`` parameter determines the cutoff for whether short
+spans of "good data" in between muscle artifacts are included in the
+surrounding "BAD" annotation.
+
+"""
+# Authors: Adonay Nunes <adonay.s.nunes@gmail.com>
+#          Luke Bloy <luke.bloy@gmail.com>
+# License: BSD (3-clause)
+
+import os.path as op
+import matplotlib.pyplot as plt
+import numpy as np
+from mne.datasets.brainstorm import bst_auditory
+from mne.io import read_raw_ctf
+from mne.preprocessing import annotate_muscle_zscore
+
+
+# Load data
+data_path = bst_auditory.data_path()
+raw_fname = op.join(data_path, 'MEG', 'bst_auditory', 'S01_AEF_20131218_01.ds')
+
+raw = read_raw_ctf(raw_fname, preload=False)
+
+raw.crop(130, 160).load_data()  # just use a fraction of data for speed here
+raw.resample(300, npad="auto")
+
+###############################################################################
+# Notch filter the data:
+#
+# .. note::
+#     If line noise is present, you should perform notch-filtering *before*
+#     detecting muscle artifacts. See :ref:`tut-section-line-noise` for an
+#     example.
+
+raw.notch_filter([50, 100])
+
+###############################################################################
+
+# The threshold is data dependent, check the optimal threshold by plotting
+# ``scores_muscle``.
+threshold_muscle = 5  # z-score
+# Choose one channel type, if there are axial gradiometers and magnetometers,
+# select magnetometers as they are more sensitive to muscle activity.
+out = annotate_muscle_zscore(raw, ch_type="mag", threshold=threshold_muscle,
+                             min_length_good=0.2, filter_freq=[110, 140])
+annot_muscle, scores_muscle = out
```

Comment:
I think it would be better to do:
```python
annot_muscle, scores_muscle = annotate_muscle_zscore(
    raw, ch_type="mag", threshold=threshold_muscle, min_length_good=0.2,
    filter_freq=[110, 140])
```

Reply:
@mmagnuski are you just looking to get Github user points by making constant comments?

Reply:
@AdoNunes questioning the good-faith motives of other contributors is not a violation of our code of conduct *per se*, but this and your other recent comment to @mmagnuski strike me as a bit impolite. If you feel that someone else is violating the [CoC](https://github.com/mne-tools/.github/blob/master/CODE_OF_CONDUCT.md), please reach out to myself, @agramfort or @larsoner.

Reply:
@AdoNunes I understand you may be a bit frustrated with the progress here, but my intentions were to finalize the PR by reading the code again and giving comments if I find any remaining issues.

Reply:
Sorry @mmagnuski, I thought that the PR was merged but I saw you made yet another comment and made me snap. IMO I felt you were constantly commenting and replying, but twice you said you were going to test the function with your eeg data, and at the end you gave me a link to test it myself. Having time available, helping out and making commits is very helpful. I wasn't going to say anything but I saw your notification when I woke up...

Reply:
Ah, sorry for that, I didn't have enough time to test the data that day so I only browsed through my files and selected one with some nice, visible artifacts. I could have made myself clearer that I could still test the PR with that data, but some other day. And I know my comments might have been too partisan-style, I get that it can be irritating, I should have waited and packed most of the comments in one go.

Reply:
@mmagnuski  thanks for your comprehension 😄